### PR TITLE
Add support populate Java Object with custom schema without FakeForSc…

### PR DESCRIPTION
**Pull request by RVRhub**
PR adds support to populate Java objects with a custom schema without using the FakeForSchema annotation. The existing implementation of `net.datafaker.providers.base.BaseFaker#populate(java.lang.Class<T>, net.datafaker.transformations.Schema<java.lang.Object,?>)` requires the use FakeForSchema annotation to generate fake data for the object.

### CODE REVIEW

File: src/main/java/net/datafaker/annotations/FakeResolver.java [hunk 1]
```diff
@@ -9,26 +9,33 @@
...
 
-        JavaObjectTransformer jTransformer = new JavaObjectTransformer();
         return (T) jTransformer.apply(clazz, useSchema);
     }
 
-    private Schema<Object, ?> getSchema(Schema<Object, ?> schema, String[] pathToSchema) {
-        if (schema != null) {
-            return schema;
-        } else if (pathToSchema.length != 0) {
+    public T generateFromCustomSchema(Schema<Object, ?> schema) {
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/annotations/FakeResolver.java:29]: why can't we keep same method name `public T generate(Schema<Object, ?> schema)` ?

File: src/main/java/net/datafaker/annotations/FakeResolver.java [hunk 2]
```diff
@@ -9,26 +9,33 @@
 
 public class FakeResolver<T> {
 
+    private static final JavaObjectTransformer jTransformer = new JavaObjectTransformer();
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/annotations/FakeResolver.java:12]: constant should be uppercase

File: src/main/java/net/datafaker/annotations/FakeResolver.java [hunk 3]
```diff
@@ -9,26 +9,33 @@
...
-        } else if (pathToSchema.length != 0) {
+    public T generateFromCustomSchema(Schema<Object, ?> schema) {
+        if (Objects.isNull(schema)) {
+            return generateFromDefaultSchema();
+        }
+
+        return (T) jTransformer.apply(clazz, schema);
+    }
+
+    private Schema<Object, T> getSchema(String[] pathToSchema) {
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/annotations/FakeResolver.java:37]: why do we need an array here if we use only the first element?
I guess just `String` should be enough

Comment (RVRhub) [reply src/main/java/net/datafaker/annotations/FakeResolver.java]: Changed value type from array to a single value.

File: src/main/java/net/datafaker/annotations/FakeResolver.java (hunk 2 above)
Comment (RVRhub) [reply src/main/java/net/datafaker/annotations/FakeResolver.java]: Renamed constant

File: src/main/java/net/datafaker/annotations/FakeResolver.java (hunk 1 above)
Comment (RVRhub) [reply src/main/java/net/datafaker/annotations/FakeResolver.java]: Made one `public T generate(Schema<Object, ?> schema)` method.

**Review by snuyanzin — APPROVED** [review]

**Comment by snuyanzin** [comment]
lgtm

**Pull request merged**